src/com/twitter/intellij/pants/ui/PantsToBspProjectAction.java

```diff
@@ -0,0 +1,192 @@
+    Project project = e.getProject();
+    if (project == null) {
+      Messages.showInfoMessage("Project not found.", "Error");
+      return;
+    }
+    dependingOnBspProjectExistence(
+      project,
+      () -> createBspProject(project),
+      linkedBspProject -> ProjectUtil.openOrImport(Paths.get(linkedBspProject), new OpenProjectTask())
+    );
+  }
+
+  private void createBspProject(Project project) {
+    ProgressManager.getInstance().run(new Task.Backgroundable(project, "Preparing BSP Project", false) {
+      @Override
+      public void run(@NotNull ProgressIndicator indicator) {
+        try {
+          GeneralCommandLine commandLine = createCommandLine(project);
+          Process process = commandLine.createProcess();
+          String output = readToString(process.getInputStream());
+          String outputErr = readToString(process.getErrorStream());
+          int result = process.waitFor();
+          if (result != 0) {
+            Path workspace = commandLine.getWorkDirectory().toPath();
+            Optional<Path> projectPath = existingProjectPath(outputErr, workspace);
+
+            projectPath.ifPresent(path -> registerNewBspProjectAndOpen(path, project));
+            projectPath.orElseThrow(() -> new RuntimeException(formatError(output, outputErr)));
+          } else {
+            Path projectPath = Paths.get(output).toAbsolutePath();
+            registerNewBspProjectAndOpen(projectPath, project);
+          }
+        }
+        catch (Exception ex) {
+          ApplicationManager.getApplication()
+            .invokeLater(() -> Messages.showErrorDialog(project, ex.getMessage(), "Error"));
+        }
+      }
+    });
+  }
+
+  private void registerNewBspProjectAndOpen(Path projectPath, Project project) {
+    PropertiesComponent.getInstance(project).setValue(BSP_LINKED_PROJECT_PATH, projectPath.toString());
+    ApplicationManager.getApplication()
+      .invokeLater(() -> ProjectUtil.openOrImport(projectPath, new OpenProjectTask()));
+  }
+
+  private Optional<Path> existingProjectPath(String output, Path workspace) {
+    Pattern pattern = Pattern.compile("can't create project named '(.*?)' because it already exists");
+    Matcher matcher = pattern.matcher(output);
+    if (matcher.find()) {
+      String name = matcher.group(1);
+      return Optional.of(workspace.resolveSibling("bsp-projects").resolve(name));
+    } else {
+      return Optional.empty();
+    }
+  }
+
+  private GeneralCommandLine createCommandLine(Project project) throws IOException {
+    GeneralCommandLine commandLine = PantsUtil.defaultCommandLine(project);
+
+    String coursier = coursierPath().toString();
+    commandLine.setExePath(coursier);
+
+    List<String> commandBase = Arrays.asList(
+      "launch", "org.scalameta:metals_2.12:latest.stable",
+      "--main", "scala.meta.internal.pantsbuild.BloopPants",
+      "--",
+      "create",
+      "--intellij",
+      "--intellijLauncher", "echo"
+    );
```

Comment:
It would be nice to log downloaded metals version or fix it in code.

Reply:
I don't think we can tell the exact Metals version from this command when using `latest.stable`. One alternative would be to additionally launch a `version` command so that we can log what version got picked up 🤔

Reply:
Right, but I think it's not essential for this PR, so it can be done as a follow-up (as well as the cousier-verison related issue). Nevertheless it must be done soon, as bug fixing may be very hard without it, in some specific cases.